### PR TITLE
Fix edge cases in metric computation

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* #3034

*Description of changes:* Excessive usage of numpy masked arrays resulted in incorrect metric value aggregation. Added extensive testing for cases where metrics are expected to evaluate to `inf` or `nan`, using different aggregations: 45/95 of the test cases added in `test_metrics.py` would fail on `dev`.

Minor changes: small refactoring and added test cases to `test_aggregations.py`.

TODOs:
- [x] add test cases

MWE:

```python
import numpy as np
from gluonts.ev.metrics import (
    ND,
    WeightedSumQuantileLoss,
    MeanWeightedSumQuantileLoss,
    MeanScaledQuantileLoss,
    AverageMeanScaledQuantileLoss,
)

data = {
    "label": np.zeros((20)),
    "0.5": np.ones(20),
    "0.1": np.ones(20),
    "0.9": np.ones(20),
    "seasonal_error": 0.0,
}


def print_with_type(v):
    print(f"{type(v)}: {v}")


metrics = [
    ND(),
    WeightedSumQuantileLoss(0.5),
    MeanWeightedSumQuantileLoss([0.1, 0.5, 0.9]),
    MeanScaledQuantileLoss(0.5),
    AverageMeanScaledQuantileLoss([0.1, 0.5, 0.9]),
]

for metric in metrics:
    print_with_type(metric().update(data).get())
```

Before:

```
<class 'numpy.ma.core.MaskedConstant'>: --
<class 'numpy.ma.core.MaskedConstant'>: --
<class 'numpy.float64'>: 0.0
/Users/stellalo/gluonts/src/gluonts/ev/stats.py:101: RuntimeWarning: divide by zero encountered in divide
  return quantile_loss(data, q) / data["seasonal_error"]
<class 'numpy.float64'>: inf
<class 'numpy.ma.core.MaskedConstant'>: --
```

After:

```
/Users/stellalo/gluonts/src/gluonts/ev/metrics.py:393: RuntimeWarning: divide by zero encountered in divide
  return sum_absolute_error / sum_absolute_label
<class 'numpy.float64'>: inf
/Users/stellalo/gluonts/src/gluonts/ev/metrics.py:472: RuntimeWarning: divide by zero encountered in divide
  return sum_quantile_loss / sum_absolute_label
<class 'numpy.float64'>: inf
<class 'numpy.float64'>: inf
/Users/stellalo/gluonts/src/gluonts/ev/stats.py:101: RuntimeWarning: divide by zero encountered in divide
  return quantile_loss(data, q) / data["seasonal_error"]
<class 'numpy.float64'>: inf
<class 'numpy.float64'>: inf
```

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

File: src/gluonts/ev/metrics.py [hunk 1]
```diff
@@ -565,7 +565,7 @@ def mean(
             [np.abs(coverages[f"coverage[{q}]"] - q) for q in quantile_levels],
```
Comment (shchur) [inline src/gluonts/ev/metrics.py:565]: I think I'm missing some context on the implementation of these methods & numpy masked array behavior to catch the potential edge cases. 

I would recommend just extensively testing the existing metric implementations, e.g., by ensuring matching values w.r.t. the old Evaluator, including cases like predictions / past data / future data are all zeros / all constant, etc.

**Comment by lostella** [comment]
@shchur added a test specifically for edge cases leading to infinity in `test_metrics.py`: of 45/95 test cases included there would fail on `dev`.

**Review by jaheba — APPROVED** [review]
